CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/SyntaxCollectionsFile.swift

```diff
@@ -0,0 +1,465 @@
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+import SwiftSyntaxBuilder
+import SyntaxSupport
+import Utils
+
+let syntaxCollectionsFile = SourceFileSyntax(leadingTrivia: [.blockComment(generateCopyrightHeader(for: "generate-swiftsyntax"))]) {
+  ProtocolDeclSyntax("""
+    public protocol SyntaxCollection: SyntaxProtocol, Sequence where Element: SyntaxProtocol {
+      /// The number of elements, `present` or `missing`, in this collection.
+      var count: Int { get }
+    }
+    """)
+  
+  ExtensionDeclSyntax("""
+    public extension SyntaxCollection {
+      static var structure: SyntaxNodeStructure {
+        return .collection(Element.self)
+      }
+    }
+    """)
+  
+  for node in SYNTAX_NODES where node.isSyntaxCollection {
+    let documentation = node.description.map { "/// " + $0 } ?? """
+      /// `\(node.name)` represents a collection of one or more
+      /// `\(node.collectionElement)` nodes. \(node.name) behaves
+      /// as a regular Swift collection, and has accessors that return new
+      /// versions of the collection with different children.
+      """
+    
+    StructDeclSyntax("""
+      \(documentation)
+      public struct \(node.name): SyntaxCollection, SyntaxHashable
+      """) {
+      if let collectionElementChoices = node.collectionElementChoices, !collectionElementChoices.isEmpty {
+        EnumDeclSyntax("""
+          @frozen // FIXME: Not actually stable, works around a miscompile
+          public enum Element: SyntaxChildChoices
+          """) {
+          for choiceName in collectionElementChoices {
+            if let choice = SYNTAX_NODE_MAP[choiceName] {
+              EnumCaseDeclSyntax("case `\(raw: choice.swiftSyntaxKind)`(\(raw: choice.name))")
+            }
+          }
+          
+          VariableDeclSyntax(
+            modifiers: [DeclModifierSyntax(name: .public)],
+            name: IdentifierPatternSyntax("_syntaxNode"),
+            type: TypeAnnotationSyntax(
+              colon: .colon,
+              type: TypeSyntax("Syntax")
+            )
+          ) {
+            SwitchStmtSyntax(switchKeyword: .switch, expression: ExprSyntax("self")) {
+              for choiceName in node.collectionElementChoices ?? [] {
+                if let choice = SYNTAX_NODE_MAP[choiceName] {
+                  SwitchCaseSyntax("case .\(raw: choice.swiftSyntaxKind)(let node):") {
+                    ReturnStmtSyntax("return node._syntaxNode")
+                  }
+                }
+              }
+            }
+          }
+          
+          InitializerDeclSyntax("init(_ data: SyntaxData) { self.init(Syntax(data))! }")
+          
+          for choiceName in node.collectionElementChoices ?? [] {
+            if let choiceNode = SYNTAX_NODE_MAP[choiceName] {
```

Comment:
If `choiceName` is not in `SYNTAX_NODE_MAP`, that seems like a pretty serious issue to me. Should we trap instead? I.e. do `choiceNode = SYNTAX_NODE_MAP[choiceName]!`

(same above and below)